homeassistant/components/hue/v2/light.py

```diff
@@ -155,13 +175,35 @@ def extra_state_attributes(self) -> dict[str, str] | None:
             "dynamics": self.resource.dynamics.status.value,
         }
 
+    @property
+    def effect(self) -> str | None:
+        """Return the current effect."""
+        if effects := self.resource.effects:
+            if effects.status != EffectStatus.NO_EFFECT:
+                return effects.status.value
+        if timed_effects := self.resource.timed_effects:
+            if timed_effects.status != TimedEffectStatus.NO_EFFECT:
+                return timed_effects.status.value
+        return EFFECT_NONE
+
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
         transition = normalize_hue_transition(kwargs.get(ATTR_TRANSITION))
         xy_color = kwargs.get(ATTR_XY_COLOR)
         color_temp = normalize_hue_colortemp(kwargs.get(ATTR_COLOR_TEMP))
         brightness = normalize_hue_brightness(kwargs.get(ATTR_BRIGHTNESS))
         flash = kwargs.get(ATTR_FLASH)
+        effect = effect_str = kwargs.get(ATTR_EFFECT)
+        if effect_str == EFFECT_NONE:
+            effect = EffectStatus.NO_EFFECT
+        elif effect_str is not None:
+            # work out if we got a regular effect or timed effect
+            effect = TimedEffectStatus(effect_str)
+            if effect == TimedEffectStatus.UNKNOWN:
+                effect = EffectStatus(effect_str)
+            if transition is None:
+                # a transition is required for timed effect, default to 10 minutes
```

Comment:
Now we also set a transition for a normal effect. Will that impact how long it takes for a light to turn on too ? 

And if a timed effect requires a transition, why not raise an error ?

Reply:
Hmm, that needs to be adjusted indeed. 

The default is there because the HA frontend doesn't allow setting a transition for the effect, there's just a dropdown

Reply:
I've adjusted the transition logic so it only applies to TimedEffect.

As for keeping the default, see my comment above: The HA frontend allows for selecting the effect in the frontend from a dropdown. Only with advanced service calls you can specify the transition yourself.